src/main/java/io/github/cottonmc/cotton/gui/widget/icon/ItemIcon.java

```diff
@@ -48,10 +47,10 @@ public void paint(MatrixStack matrices, int x, int y, int size) {
 
 		float scale = size != 16 ? ((float) size / 16f) : 1f;
 
-		RenderSystem.pushMatrix();
-		RenderSystem.translatef(x, y, 0);
-		RenderSystem.scalef(scale, scale, 1);
+		matrices.push();
+		matrices.translate(x, y, 0);
+		matrices.scale(scale, scale, 1);
 		renderer.renderInGui(stack, 0, 0);
-		RenderSystem.popMatrix();
+		matrices.pop();
```

Comment:
I'm not sure if this actually works as the `renderInGui` call does not take the matrices, so it can't use the matrix transformations from there.